docs/sources/reference/components/otelcol/otelcol.processor.groupbyattrs.md

```diff
@@ -0,0 +1,88 @@
+# otelcol.processor.groupbyattrs
+
+`otelcol.processor.groupbyattrs` accepts telemetry data from other `otelcol`
+components and reassociates spans, log records and metric datapoints to a Resource that matches with the specified attributes. It groups telemetry data by specified attributes.
+
+{{% admonition type="note" %}}
+`otelcol.processor.groupbyattrs` is a wrapper over the upstream OpenTelemetry
+Collector `groupbyattrs` processor.  If necessary, bug reports or feature requests
+will be redirected to the upstream repository.
+{{% /admonition %}}
+
+It is recommended to use the groupbyattrs processor together with batch processor, as a consecutive step, as this will reduce the fragmentation of data (by grouping records together under matching Resource/Instrumentation Library)
+
+You can specify multiple `otelcol.processor.groupbyattrs` components by giving them
+different labels.
+
+## Usage
+
+```river
+otelcol.processor.groupbyattrs "LABEL" {
+  output {
+    metrics = [...]
+    logs    = [...]
+    traces  = [...]
+  }
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type            | Description                                                                           | Default    | Required
+---- |-----------------|---------------------------------------------------------------------------------------|------------| --------
+`keys` | `array(string)` | Keys that will be used to group the spans, log records or metric data points together | | no
+
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name    | Type               | Description
+--------|--------------------|-----------------------------------------------------------------
+`input` | `otelcol.Consumer` | A value that other components can use to send telemetry data to.
+
+`input` accepts `otelcol.Consumer` data for any telemetry signal (metrics,
+logs, or traces).
+
+## Component health
+
+`otelcol.processor.groupbyattrs` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`otelcol.processor.groupbyattrs` does not expose any component-specific debug
+information.
+
+## Debug metrics
+
+`otelcol.processor.groupbyattrs` does not expose any component-specific debug metrics.
+
+## Examples
+
+### Grouping metrics by an attribute
+
+This example reassociates the metrics based on the value of the `host.name` attribute.
+
+```alloy
+otelcol.processor.groupbyattrs "default" {
+  keys = [
+    "host.name",
+  ]
+
+  output {
+    metrics = [otelcol.exporter.otlp.default.input]
+    logs    = [otelcol.exporter.otlp.default.input]
+    traces  = [otelcol.exporter.otlp.default.input]
+  }
+}
+```
+
+## Notes
```

Comment:
Non-blocking commen.
I'm not sure that this info is best grouped under a `Notes` heading.  I'll leave it as-is... just noting it here in the review that this is not used elsewhere, and it might be better to rethink how this information is presented. It's something we can put off to a future PR.

Reply:
I moved the last note as part of the documentation for the Keys attribute